modules/miner.go

```diff
@@ -66,8 +70,4 @@ type CPUMiner interface {
 type Miner interface {
 	BlockManager
 	CPUMiner
```

Comment:
I'm confused by the various `Miner` types now. If we have a `TestMiner`, why is `Miner` also a `CPUMiner`? Isn't the `TestMiner` a CPU Miner? Is CPU mining even allowed/possible anymore? If not, shouldn't the `Miner` just be a `BlockManager`?